metadata/metadata.go

```diff
@@ -188,7 +188,7 @@ func FromIncomingContext(ctx context.Context) (MD, bool) {
 		// map, and there's no guarantee that the MD attached to the context is
 		// created using our helper functions.
 		key := strings.ToLower(k)
-		out[key] = v
+		out[key] = append(out[key], v...)
```

Comment:
Would it be a tiny bit more performant to use `append(([]string)(nil), v...)` instead (avoiding a map lookup)?

Reply:
Done